src/runtime/crdt/crdt-collection.ts

```diff
@@ -113,21 +175,32 @@ export class CRDTCollection<T extends Referenceable> implements CollectionModel<
     return true;
   }
 
-  private mergeItems(data1: CollectionData<T>, data2: CollectionData<T>): Dictionary<{value: T, version: VersionMap}> {
-    const merged: Dictionary<{value: T, version: VersionMap}> = {};
-    for (const {value, version: version2} of Object.values(data2.values)) {
-      if (this.model.values[value.id]) {
-        merged[value.id] = {value, version: mergeVersions(this.model.values[value.id].version, version2)};
-      } else if (!dominates(data1.version, version2)) {
-        merged[value.id] = {value, version: version2};
+  private fastForward(op: CollectionFastForwardOp<T>): boolean {
+    const currentClock = this.model.version;
+    if (!dominates(currentClock, op.oldClock)) {
+      // Can't apply fast-forward op. Current model's clock is behind oldClock.
+      return false;
+    }
+    if (dominates(currentClock, op.newClock)) {
+      // Current model already knows about everything in this fast-forward op.
+      return false;
```

Comment:
Should this return true? This is a no-op, which could be viewed as a success-by-default.